src/main/java/com/checkmarx/flow/service/ThresholdValidatorImpl.java

```diff
@@ -53,30 +54,58 @@ public ThresholdValidatorImpl(@Lazy SastScanner sastScanner, @Lazy SCAScanner sc
     public boolean isMergeAllowed(ScanResults scanResults, RepoProperties repoProperties, PullRequestReport pullRequestReport) {
         OperationResult requestResult = new OperationResult(OperationStatus.SUCCESS, MERGE_SUCCESS_DESCRIPTION);
-        boolean isMergeAllowed = isAllowed(scanResults, repoProperties, pullRequestReport);
+        boolean isMergeAllowed = true;
 
-        if (!isMergeAllowed) {
-            requestResult = new OperationResult(OperationStatus.FAILURE, MERGE_FAILURE_DESCRIPTION);
+        if (!repoProperties.isErrorMerge()) {
+            log.info("Merge is allowed, because error-merge is set to false.");
+            pullRequestReport.setPullRequestResult(requestResult);
+        }
+        else{
+            isMergeAllowed = isAllowed(scanResults, pullRequestReport.getScanRequest());
+
+            if (!isMergeAllowed) {
+                log.info("Merge is not allowed, because some thresholds were exceeded.");
+                requestResult = new OperationResult(OperationStatus.FAILURE, MERGE_FAILURE_DESCRIPTION);
+            }
         }
 
         pullRequestReport.setPullRequestResult(requestResult);
 
         return isMergeAllowed;
     }
 
-    private boolean isAllowed(ScanResults scanResults, RepoProperties repoProperties, PullRequestReport pullRequestReport) {
-        if (!repoProperties.isErrorMerge()) {
-            log.info("Merge is allowed, because error-merge is set to false.");
-            return true;
+    @Override
+    public boolean thresholdsExceeded(ScanRequest request, ScanResults results){
+        return !isAllowed(results, request);
+    }
+
+    @Override
+    public boolean isThresholdsConfigurationExist(ScanRequest scanRequest){
+        boolean sastThresholds;
+        boolean scaThresholds = false;
+
+        sastThresholds = scanRequest.getThresholds() != null || flowProperties.getThresholds() != null;
+        if(scaProperties != null){
+            scaThresholds = scaProperties.getThresholdsSeverity() != null || scaProperties.getThresholdsScore() != null;
+        }
+
+        if (!scaThresholds && scanRequest.getScaConfig() != null) {
+            scaThresholds = scanRequest.getScaConfig().getThresholdsSeverity() != null || scanRequest.getScaConfig().getThresholdsScore() != null;
         }
 
+        log.info("Checking Thresholds exists. sast thresholds: {}. sca thresholds: {}", sastThresholds, scaThresholds);
+        return  sastThresholds || scaThresholds;
+    }
+
+    private boolean isAllowed(ScanResults scanResults, ScanRequest request) {
+
         boolean isAllowed = true;
         if (isSast()) {
-            isAllowed = isAllowedSast(scanResults, pullRequestReport);
+            isAllowed = isAllowedSast(scanResults, request);
         }
 
         if (isAllowed && scaScanner.isEnabled()) {
-            isAllowed = isAllowedSca(scanResults, pullRequestReport);
+            isAllowed = isAllowedSca(scanResults, request);
         }
 
         return isAllowed;
```

Comment:
this looks more readable to me:
```java
boolean isSastAllowed = isSast() ? isAllowedSast(scanResults, request) : true;
boolean isScaAllowed = scaScanner.isEnabled() ?  isAllowedSca(scanResults, request) : true;
return isSastAllowed && isScaAllowed 
```

Reply:
The result would be different. we want that if one of the scanner is enabled and its thresholds were exceeded - to return false (not allowed)
in your case, if SAST not enabled (only SCA), and SCA thresholds not exceeded - still we will return notAllowed

Reply:
if SAST not enabled -> isSast() == false -> boolean isSastAllowed = true
SCA thresholds not exceeded -> scaScanner.isEnabled() == true -> isAllowedSca(scanResults, request) -> true
return isSastAllowed && isScaAllowed -> true && true -> true
still we will return notAllowed <- no as you can see we will return true

Reply:
You are right I changed that (and simplified per intelij suggestion)